synapse/app/generic_worker.py

```diff
@@ -738,6 +738,11 @@ async def _process_and_notify(self, stream_name, instance_name, token, rows):
         except Exception:
             logger.exception("Error processing replication")
 
+    async def on_position(self, stream_name: str, instance_name: str, token: int):
+        await super().on_position(stream_name, instance_name, token)
+        if stream_name == TypingStream.NAME:
+            self.typing_handler.process_replication_rows(token, [])
```

Comment:
This will need to be expanded to include additional streams, if it is the right fix.